pageserver/src/tenant/ephemeral_file.rs

```diff
@@ -74,58 +79,18 @@ impl EphemeralFile {
         l.files.insert(file_id, file_rc.clone());
 
         Ok(EphemeralFile {
-            file_id,
+            ephemeral_files_id: file_id,
+            page_cache_file_id: page_cache::next_file_id(),
             _tenant_id: tenant_id,
             _timeline_id: timeline_id,
             file: file_rc,
             size: 0,
+            mutable_tail: [0u8; PAGE_SZ],
         })
     }
 
-    fn fill_buffer(&self, buf: &mut [u8], blkno: u32) -> Result<(), io::Error> {
-        let mut off = 0;
-        while off < PAGE_SZ {
-            let n = self
-                .file
-                .read_at(&mut buf[off..], blkno as u64 * PAGE_SZ as u64 + off as u64)?;
-
-            if n == 0 {
-                // Reached EOF. Fill the rest of the buffer with zeros.
-                const ZERO_BUF: [u8; PAGE_SZ] = [0u8; PAGE_SZ];
-
-                buf[off..].copy_from_slice(&ZERO_BUF[off..]);
-                break;
-            }
-
-            off += n;
-        }
-        Ok(())
-    }
-
-    fn get_buf_for_write(
-        &self,
-        blkno: u32,
-    ) -> Result<page_cache::PageWriteGuard<'static>, io::Error> {
-        // Look up the right page
-        let cache = page_cache::get();
-        let mut write_guard = match cache
-            .write_ephemeral_buf(self.file_id, blkno)
-            .map_err(|e| to_io_error(e, "Failed to write ephemeral buf"))?
-        {
-            WriteBufResult::Found(guard) => guard,
-            WriteBufResult::NotFound(mut guard) => {
-                // Read the page from disk into the buffer
-                // TODO: if we're overwriting the whole page, no need to read it in first
-                self.fill_buffer(guard.deref_mut(), blkno)?;
-                guard.mark_valid();
-
-                // And then fall through to modify it.
-                guard
-            }
-        };
-        write_guard.mark_dirty();
-
-        Ok(write_guard)
+    pub(crate) fn size(&self) -> u64 {
+        self.size
```

Comment:
What's the benefit of this? If we are going to change this, I think it should follow https://doc.rust-lang.org/stable/std/fs/struct.Metadata.html#method.len or https://doc.rust-lang.org/stable/std/fs/struct.File.html#method.set_len naming as they are the same, as far as I understand.

Reply:
It's to make the EphemeralSize::size field private.
Because it's super relevant for correctness of `write_blob`.

Nobody cares about EphemeralFile API looking similar to `std::fs::File`.
In fact that'd be misleading, as EphemeralFile can only be written to using `write_blob` and only read from through `read_blob`.

Reply:
> EphmeralSize::size

EphmeralFile::size.

But it's still a file with len. Small streams of mental overhead => rivers where to drown.

Reply:
It's a pre-existing API, I have no interest in changing it in this PR, feel free to to open a follow-up one.

Reply:
I also think that len is better than size but I agree, it should be in a future PR.

Reply:
#5034